src/main/java/io/cdap/plugin/salesforce/plugin/source/batch/base/SalesforceBaseBatchSource.java

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.cdap.plugin.salesforce.plugin.source.batch.base;
+
+import com.google.common.base.Strings;
+import io.cdap.cdap.api.data.format.StructuredRecord;
+import io.cdap.cdap.api.data.format.UnexpectedFormatException;
+import io.cdap.cdap.api.data.schema.Schema;
+import io.cdap.cdap.api.dataset.lib.KeyValue;
+import io.cdap.cdap.etl.api.Emitter;
+import io.cdap.cdap.etl.api.InvalidEntry;
+import io.cdap.cdap.etl.api.batch.BatchSource;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Base class for Salesforce Batch Source plugins. Contains data transformation logic.
+ *
+ * @param <T> input data key value
+ */
+public abstract class SalesforceBaseBatchSource<T> extends BatchSource<T, Map<String, String>, StructuredRecord> {
```

Comment:
Instead of an abstract base plugin class, this logic should be pulled out into a class that transforms a single line from the bulk API response to a StructuredRecord.